### ISSUE TITLE
in master, you cant open both server and client the same time

### ISSUE THREAD

**Issue by est31** [minetest/minetest#2344]
Try starting a server, and then start a client, you'll get:
```
4: Bind failed: Die Adresse wird bereits verwendet
terminate called after throwing an instance of 'con::ConnectionBindFailed'
  what():  Failed to bind socket (port already in use?)
Abgebrochen
```
Starting two clients in parallel works fine. Workaround: give client an unused port via `--port` option.
Bisected it and ed04e8e9e407f0dd57fa83a9732b3a3968cb80e0 is the first bad commit. @nerzhul 
Guess the root cause is that client now tries to listen on port, but closes port again when it realizes its a client and no server.

**Comment by est31** [comment]
As it seems, its just me expecting server to be fast. Perhaps there is a startup time decrease, I'm not sure, but this is sth for another issue.
